Xamarin.Forms.FlexLayout/Xamarin.FlexLayout.Shared/Xamarin.Forms.FlexLayout.cs

```diff
@@ -0,0 +1,313 @@
+			_root.FlexDirection = FlexDirection;
+			_root.JustifyContent = JustifyContent;
+			_root.AlignContent = AlignContent;
+			_root.AlignItems = AlignItems;
+			_root.Overflow = Overflow;
+			_root.PositionType = Position;
+			_root.Wrap = Wrap;
+			_root.AlignSelf = GetAlignSelf(this);
+			_root.FlexGrow = (float)GetGrow(this);
+			_root.FlexBasis = (float)GetBasis(this);
+			_root.FlexShrink = (float)GetShrink(this);
+		}
+
+		protected override void OnPropertyChanged([CallerMemberName] string propertyName = null)
+		{
+			base.OnPropertyChanged(propertyName);
+			UpdateNode(this, this, propertyName);
+		}
+
+		void AttachNodesFromViewHierachy(View view)
+		{
+			var node = GetNode(view);
+			// Only leaf nodes should have a measure function
+			if (view.IsLeaf())
+			{
+				node.Clear();
+				node.SetMeasure(MeasureView);
+				return;
+			}
+
+			node.SetMeasure(null);
+
+			// Create a list of all the subviews that we are going to use for layout.
+			var subviewsToInclude = new List<View>();
+			foreach (var subview in FlexLayoutExtensions.GetChildren(view))
+			{
+				if (GetIsIncluded(subview))
+				{
+					subviewsToInclude.Add(subview);
+				}
+			}
+
+			if (!NodeHasExactSameChildren(node, subviewsToInclude.ToArray()))
+			{
+				node.Clear();
+				for (int i = 0; i < subviewsToInclude.Count; i++)
+				{
+					var subView = subviewsToInclude[i];
+					var subViewNode = GetNode(subView);
+					subViewNode.MarginLeft = (float)subView.Margin.Left;
+					subViewNode.MarginTop = (float)subView.Margin.Top;
+					subViewNode.MarginRight = (float)subView.Margin.Right;
+					subViewNode.MarginBottom = (float)subView.Margin.Bottom;
+					subViewNode.FlexGrow = (float)GetGrow(subView);
+					subViewNode.FlexBasis = (float)GetBasis(subView);
+					subViewNode.FlexShrink = (float)GetShrink(subView);
+					subViewNode.AlignSelf = GetAlignSelf(subView);
+					node.Insert(i, subViewNode);
+				}
+			}
+
+			foreach (var subView in subviewsToInclude)
+				AttachNodesFromViewHierachy(subView);
+
+		}
+
+		Size MeasureView(IFlexNode node, float width, float height)
+		{
+			View view = node.Data as View;
+
+			var sizeRequest = view.Measure(width, height);
+
+			return sizeRequest.Request;
+		}
+
+		static void UpdateNode(FlexLayout parent, object element, string propertyName)
+		{
+			bool shouldInvalidate = false;
+			if (propertyName.Equals(WidthRequestProperty.PropertyName) ||
+				propertyName.Equals(HeightRequestProperty.PropertyName) ||
+				propertyName.Equals(MinimumHeightRequestProperty.PropertyName) ||
+				propertyName.Equals(MinimumWidthRequestProperty.PropertyName))
+			{
+				var view = element as View;
+				if (view == null)
+					throw new ArgumentNullException(nameof(element));
+
+				var node = GetNode(view);
+				node.Width = (float)view.WidthRequest;
+				node.Height = (float)view.HeightRequest;
+				shouldInvalidate = true;
+			}
+
+			if (propertyName.Equals(_FlexAlignSelfPropertyName) ||
```

Comment:
I'd do `FlexAlignSelfProperty.PropertyName`, and suddenly that `const` is no longer required at all